concise_concepts/conceptualizer/Conceptualizer.py

```diff
@@ -357,17 +329,18 @@ def add_patterns(input_dict):
                     words = input_dict[key]
                 for word in words:
                     if word != key:
-                        specific_match_rule = dict()
-                        specific_match_rule.update(self.match_rule)
-                        word_parts = re.split(f"[{self.word_delimiter}]+", word)
+                        specific_match_rule = {**self.match_rule}
+                        word_parts = re.split(
+                            f"[{re.escape(self.word_delimiter)}]+", word
+                        )
                         if len(word_parts) > 1:
                             operators = [" ", "-"]
                         else:
                             operators = [""]
 
                         for op in operators:
                             if self.case_sensitive:
-                                specific_match_rule[self.match_key] = "{op}".join(
+                                specific_match_rule[self.match_key] = f"{op}".join(
```

Comment:
One of the critical issues described in the PR.

Reply:
god, so sloppy😅